csharp-basic-topics/HowToConvertReadOnlyMemoryToByteArrayInCSharp/HowToConvertReadOnlyMemoryToByteArrayInCSharp/Program.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        const string text = "foo";
+
+        var byteArray = ReadOnlyMemoryToByteArray<int>(new[] { 1, 2, 3 });
+        Console.WriteLine(BitConverter.ToString(byteArray));
+
+        byteArray = ReadOnlyMemoryToByteArray(text.AsMemory());
+        Console.WriteLine(BitConverter.ToString(byteArray));
+
+        var path = Path.GetTempFileName();
+        byteArray = SaveText(path, text.AsMemory());
+        Console.WriteLine(BitConverter.ToString(byteArray));
+
+        File.Delete(path);
+    }
+
+    public static byte[] ReadOnlyMemoryToByteArray<T>(ReadOnlyMemory<T> input)
+        where T : struct
+    {
+        return MemoryMarshal.AsBytes(input.Span).ToArray();
+    }
+
+    public static byte[] SaveText(string path, ReadOnlyMemory<char> text)
+    {
+        using var hashAlgorithm = SHA256.Create();
```

Comment:
No need to create an instance of the hasher, we can just use the static SHA256.HashData, HashDataAsync